ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,266 @@
+        shadowDepth = appearanceProxy.shadowDepth
+    }
+}
+
+// Mark - Drawer View
+
+@objc public enum DrawerDirection: Int, CaseIterable {
+    case left
+    case right
+}
+
+/**
+ `Drawer` is used to present a content that is currently in hideout mode when collapsed and when expanded is used to present content view partially either on the left or right.
+ 
+ Currently `Drawer` only support horizontal axis and is expanded by default from left side of the screen unless explicitly specified
+ 
+ Set `Content` to provide content for the drawer.
+ */
+public struct Drawer<Content: View>: View {
+
+    @ObservedObject var state: DrawerState
+    @ObservedObject var tokens: DrawerTokens
+
+    private let backgroundLayerColor: Color = .black
+    private var backgroundLayerOpacity: Double {
+        return state.backgroundDimmed ? 0.5 : 0
+    }
+    private let backgroundLayerAnimation: Animation = .default
+
+    private let percentWidthOfContent: CGFloat = 0.9
+    private let percentSnapWidthOfScreen: CGFloat = 0.225
+
+    // content view on top of
+    public var content: Content
+
+    // `ivar` to keep track of dragged offset
+    @State private var draggedOffsetWidth: CGFloat?
+    private func computedOffset(screenSize: CGSize) -> CGFloat {
+
+        if let draggedOffsetWidth = draggedOffsetWidth {
+            return draggedOffsetWidth
+        }
+
+        var width = CGFloat.zero
+        let contentWidth = screenSize.width * percentWidthOfContent
+        if !state.isExpanded { // if closed then hidden behind the screen
+            width = state.presentationDirection == .left ? -contentWidth : contentWidth
+        }
+
+        return width
+    }
+
+    public init(content: Content) {
+        self.content = content
+        tokens = DrawerTokens()
+        state = DrawerState()
+    }
+
+    public var body: some View {
+        GeometryReader { reader in
+            HStack {
+                if state.presentationDirection == .right {
+                    Spacer()
+                }
+                content
+                    .frame(width: reader.size.width * percentWidthOfContent)
+                    .shadow(color: tokens.shadowColor.opacity(state.isExpanded ? tokens.shadowOpacity : 0),
+                            radius: tokens.shadowBlur,
+                            x: tokens.shadowDepth[0],
+                            y: tokens.shadowDepth[1])
+                    .offset(x: computedOffset(screenSize: reader.size))
+                    .animation(backgroundLayerAnimation)
+
+                if state.presentationDirection == .left {
+                    Spacer()
+                }
+            }
+            .background(state.isExpanded ? backgroundLayerColor.opacity(backgroundLayerOpacity) : Color.clear)
+            .gesture(dragGesture(snapWidth: reader.size.width * percentSnapWidthOfScreen))
+        }
+    }
+
+    private func dragGesture(snapWidth: CGFloat) -> some Gesture {
```

Comment:
Another issue you can file as a GitHub issue to be addressed after your first merge:

The current drawer presents the contents as you drag from the side instead of detecting the drag and presenting it entirely:


https://user-images.githubusercontent.com/68076145/102935613-585af200-445b-11eb-8a9c-99b48f8a66a7.mov

Reply:
I have created a new issue to track this. 
https://github.com/microsoft/fluentui-apple/issues/357